scripts/make/sm-k6.mk

```diff
@@ -11,11 +11,17 @@ $(DISTDIR)/$(1)-$(2)/sm-k6:
 	curl -sSL https://github.com/grafana/xk6-sm/releases/download/v0.0.3/sm-k6-$(1)-$(2) -o "$$@"
 	chmod +x "$$@"
 
-sm-k6: $(DISTDIR)/$(1)-$(2)/sm-k6
+$(DISTDIR)/$(1)-$(2)/sm-k6-gsm:
+	mkdir -p "$(DISTDIR)/$(1)-$(2)"
+	# Renovate updates the following line. Keep its syntax as it is.
+	curl -sSL https://github.com/grafana/xk6-sm/releases/download/v0.4.0/sm-k6-$(1)-$(2)-gsm -o "$$@"
+	chmod +x "$$@"
+
+sm-k6: $(DISTDIR)/$(1)-$(2)/sm-k6 $(DISTDIR)/$(1)-$(2)/sm-k6-gsm
```

Comment:
In practice, this will only be used by the pipeline that packages the agent for Debian/Fedora (and derivatives). Container images rely on the `ADD` instruction (which you added).

However, since you haven't added that file to nfpm: https://github.com/grafana/synthetic-monitoring-agent/blob/ad3ffbbecf44565f74aefedc5dd3a91b6d138ad2/scripts/package/nfpm.yaml.template#L23

This means this is effectively ignored. If we do not intend to ship this in linux packages, I'd suggest getting rid of the make-related changes. If we do want to ship this binary in linux packages, then we should edit the nfpm config.

Reply:
Good spot, thanks. Adding this to images only is probably enough for now.

Reply:
Although, that could become problematic so its probably better to add it everywhere 🤔

Reply:
Updated, hopefully I understood correctly